third_party/move/move-compiler-v2/tests/checking/typing/type_variable_join_threaded_unpack_assign_invalid.exp

```diff
@@ -0,0 +1,7 @@
+
+Diagnostics:
+error: `integer` not compatible with `bool` in name expression
+   ┌─ tests/checking/typing/type_variable_join_threaded_unpack_assign_invalid.move:36:9
+   │
+36 │         f1
```

Comment:
Sorry for being redundant, but it occurs to me that what we need is to do type checking bottom-up, so that "f1" is fine, but when we see the context, we can say `integer` not compatible with `bool` in sequence expression value.  or maybe you need something even term-specific.

Reply:
I do not agree with you, see above. Actually that an expected type is pushed from the context into the checker is a feature which is very useful and leads to better error messages in many cases.

We can also implement an alternative type checker which works bottom up and compare, but this is likely a month of work or more...